elastic/kibana
# [ResponseOps][Alerts] Migrate alerts fetching to TanStack Query

## Summary

Implements a new `useSearchAlertsQuery` hook based on TanStack Query to replace the `useFetchAlerts` hook, following [this organizational logic](https://github.com/elastic/kibana/issues/186448#issuecomment-2228853337).

This PR focuses mainly on the fetching logic itself, leaving the surrounding API surface mostly unchanged since it will be likely addressed in subsequent PRs.

## To verify

1. Create rules that fire alerts in different solutions
2. Check that the alerts table usages work correctly ({O11y, Security, Stack} alerts and rule details pages, ...)
    1. Check that the alerts displayed in the table are coherent with the solution, KQL query, time filter, pagination
    2. Check that pagination changes are reflected in the table
    3. Check that changing the query when in pages > 0 resets the pagination to the first page

Closes point 1 of https://github.com/elastic/kibana/issues/186448
Should fix https://github.com/elastic/kibana/issues/171738

### Checklist
